### DISCUSSION TITLE
Possible to do implicit conversion of number to string?

### DISCUSSION THREAD

**Discussion by jezeke** [nlohmann/json#2249] [Q&A] [answered]
Doing the following, I would expect that the number could be implicitly converted to a string, however instead a type error exception
 is thrown.

```
j["a"] = 5;
std::string a = j["a"].get<std::string>();
```

My use-case is an API which accepts values in base64 ``{key: "aE2f"}``, and in some cases of purely numeric values some clients send these as numbers ``{key: 1234}``, rather than strings ``{key: "1234"}``. Is there a better solution than doing something like below each time a string value may run the risk of being sent as a number?

```
std::string a;

if(j["a"].is_string())
{
  a = j["a"].get<std::string>();
}
else
{
  a = std::to_string(j["a"].get<int>();
}
```

 I haven't done comprehensive tests but I worry I'll need to do this any time I take a string value from JSON being sent to my server, just incase the input includes purely numeric characters and their client is loosely typed and sometimes implicitly casts it to integer. Any help or info would be greatly appreciated.

**Comment by gregmarr** [comment, marked as answer]
If you don't know the types of the values coming in, then you do need to check the types, as you've written.  If you're going to do this a lot, create a helper function for this.

    std::string get_number_as_string(json const &j) {
        if(j.is_string()) {
           return j.get<std::string>();
        }

        return std::to_string(j.get<int>());
    }

    auto a = get_number_as_string(j["a"]);

**Reply by jezeke** [reply]
Yup of course factoring out is the best solution if I'm having to do this regularly. Thought I should check there's no simpler, wiser, or inbuilt solution to do this (and to save me having to go through and swap out all my ``.get<std::string>()`` calls for a helper function... oh well).

Cheers for the response.
